server/pulp/server/webservices/views/util.py

```diff
@@ -61,7 +82,8 @@ def wrapper(*args, **kwargs):
             return func(*args, **kwargs)
 
         try:
-            request.body_as_json = json.loads(request.body)
+            request_json = json.loads(request.body)
+            request.body_as_json = _ensure_input_encoding(request_json)
```

Comment:
All the points made here are valid, but are not relevant to this line. `_ensure_input_encoding` can raise, but not a ValueError, so that line should be moved.